### ISSUE TITLE
What if the schema is behind Authorization?

### ISSUE THREAD

**Issue by Thorium** [fsprojects/SwaggerProvider#148]
I initialise the provider as in the documentations:
```fsharp
let [<Literal>] Schema = "..."
type myProvider = SwaggerProvider.OpenApiClientProvider<Schema>
```

I get a not-very descriptive error: `The type provider 'SwaggerProvider.OpenApi ClientTypeProvider' reported an error: OpenAPI specification version '' is not supported.`

The reason for this is that the API needs Authorization header, and the schema is also behind the authentication, so the `swagger.json` says:

```json
{
    "error": "Authorization field missing"
}
```

Umm...ok. What should I do now? 

1. Can I add authorization header to the initial schema query?
2. Or should I download the schema.json and refer a local file? Can I do schema like `../sample/mySchema.json` referring to a local folder and not a network path?

**Comment by sergey-tihon** [comment]
> Can I add authorization header to the initial schema query?

No, you cannot. Any real auth header expire in X hour and you will not be able to compile any more.

> Or should I download the schema.json and refer a local file?

Yes, you are able to use local path
``` fsharp
let [<Literal>] Schema = __SOURCE_DIRECTORY__ + "/../Schemas/v2/petstore.json"
type PetStore = SwaggerProvider.OpenApiClientProvider<Schema, PreferAsync = true>
```

> get a not-very descriptive error: 

Agree, it looks strange, but also strange that instead of `401` status code your backend return `200 OK` and json with `error` message.
I believe that it will be more intuitive in the case of `401` response. 

In you case, I requested schema, I received successful response with valid JSON and then [OpenAPI.Net](https://github.com/Microsoft/OpenAPI.NET) failed to parse this json schema. So I provided you back (not very useful) diagnostic information.

https://github.com/fsprojects/SwaggerProvider/blob/master/src/SwaggerProvider.DesignTime/Provider.OpenApiClient.fs#L73-L78
